bundles/org.openhab.binding.echonetlite/src/main/java/org/openhab/binding/echonetlite/internal/EchonetDevice.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.echonetlite.internal;
+
+import static org.openhab.binding.echonetlite.internal.HexUtil.hex;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Michael Barker - Initial contribution
+ */
+@NonNullByDefault
+public class EchonetDevice extends EchonetObject {
+
+    private final LinkedHashMap<Epc, State> pendingSets = new LinkedHashMap<>();
+    private final HashMap<Epc, State> stateFields = new HashMap<>();
+    private final HashMap<String, Epc> epcByChannelId = new HashMap<>();
+    private final Logger logger = LoggerFactory.getLogger(EchonetDevice.class);
+    @Nullable
+    private EchonetPropertyMap getPropertyMap;
+    private EchonetDeviceListener listener;
+    private boolean initialised = false;
+
+    private long lastPollMs = 0;
```

Comment:
Consider using `Instant` instead of milliseconds from clock to be less susceptible to changes in system clock.

Reply:
I can understand the need to use a monotonic time source here, but I'm not sure that `Instant` provides that.  I'm also loath to use `Instant` do to the polling frequency and the amount of garbage it will create.  Instead I'm going to derive a timestamp from `System.nanoTime()` which is monotonic where supported (which will be most OSs).  The timestamp is only used for relative timing (i.e. timeouts), so a actual time (e.g. since epoch) is not required.